### PR TITLE
[20129] Bugfix: fix feature/xtypes1.3 branch build

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
## Description

This PR:

* regenerates types using the Fast DDS-Gen intermediate branch.
* Due to change in enum generated code, change in MonitorService IDL file to use const instead of enum.

## Contributor Checklist
- [ ] Commit messages follow the project guidelines. 
- [ ] The code follows the style guidelines of this project. 
- [ ] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [ ] Any new/modified methods have been properly documented using Doxygen. 
- [ ] Changes are ABI compatible. 
- [ ] Changes are API compatible. 
- [ ] New feature has been added to the `versions.md` file (if applicable).
- [ ] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [ ] Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

File: include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobjectCdrAux.hpp [hunk 1]
```diff
@@ -24,13 +24,13 @@
 
 #include "dds_xtypes_typeobject.hpp"
 
-constexpr uint32_t eprosima_fastdds_dds_xtypes_CompleteStructMember_max_cdr_typesize {1464UL};
+constexpr uint32_t eprosima_fastdds_dds_xtypes_CompleteStructMember_max_cdr_typesize {1436UL};
```
Comment (richiware) [inline include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobjectCdrAux.hpp:27]: Do you know what is provoking this change?

File: src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp [hunk 2]
```diff
@@ -154,7 +154,7 @@ eProsima_user_DllExport void serialize(
     eprosima::fastcdr::Cdr::state current_state(scdr);
     scdr.begin_serialize_type(current_state,
             eprosima::fastcdr::CdrVersion::XCDRv2 == scdr.get_cdr_version() ?
-            eprosima::fastcdr::EncodingAlgorithmFlag::DELIMIT_CDR2 :
+            eprosima::fastcdr::EncodingAlgorithmFlag::PLAIN_CDR2 :
```
Comment (richiware) [inline src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp:157]: why this is final now? I don't see changes in the IDL file.

File: src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp [hunk 3]
```diff
@@ -196,6 +196,10 @@ eProsima_user_DllExport void deserialize(
                         {
                                                         case EK_COMPLETE:
                                                         case EK_MINIMAL:
+                                                            if (mid != 1)
```
Comment (richiware) [inline src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp:199]: This is an issue fixed currently in fastddsgen master branch. We should remove it from this code.

File: src/cpp/statistics/types/monitorservice_typesCdrAux.hpp [hunk 4]
```diff
@@ -37,24 +37,24 @@ constexpr uint32_t eprosima_fastdds_statistics_DeadlineMissedStatus_s_max_cdr_ty
...
 
 
+
 constexpr uint32_t eprosima_fastdds_statistics_QosPolicyCount_s_max_cdr_typesize {12UL};
 constexpr uint32_t eprosima_fastdds_statistics_QosPolicyCount_s_max_key_cdr_typesize {0UL};
 
 
-
-constexpr uint32_t eprosima_fastdds_statistics_MonitorServiceStatusData_max_cdr_typesize {565252UL};
+constexpr uint32_t eprosima_fastdds_statistics_MonitorServiceStatusData_max_cdr_typesize {72UL};
```
Comment (richiware) [inline src/cpp/statistics/types/monitorservice_typesCdrAux.hpp:45]: What a big decrease in the number!!

File: test/blackbox/types/HelloWorldTypeObject.cxx [hunk 5]
```diff
@@ -115,7 +113,7 @@ const TypeObject* GetMinimalHelloWorldObject()
 
 
     MD5 index_hash("index");
-    for (int i = 0; i < 4; ++i)
+    for(int i = 0; i < 4; ++i)
```
Comment (richiware) [inline test/blackbox/types/HelloWorldTypeObject.cxx:116]: This is not from our our c++ style code

File: test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp [hunk 6]
```diff
@@ -600,12 +600,12 @@ class DomainParticipantImpl
 
     ReturnCode_t enable_monitor_service() const
     {
-        return ReturnCode_t{ReturnCode_t::RETCODE_OK};
+        return ReturnCode_t{RETCODE_OK};
```
Comment (richiware) [inline test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp:603]:
```suggestion
        return RETCODE_OK;
```

File: test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp [hunk 7]
```diff
@@ -600,12 +600,12 @@ class DomainParticipantImpl
...
     }
 
     ReturnCode_t disable_monitor_service() const
     {
-        return ReturnCode_t{ReturnCode_t::RETCODE_OK};
+        return ReturnCode_t{RETCODE_OK};
```
Comment (richiware) [inline test/mock/dds/DomainParticipantImpl/fastdds/domain/DomainParticipantImpl.hpp:608]:
```suggestion
        return RETCODE_OK;
```

File: test/mock/rtps/BuiltinProtocols/fastdds/rtps/builtin/BuiltinProtocols.h [hunk 8]
```diff
@@ -76,8 +77,8 @@ class BuiltinProtocols
     //!Pointer to the WLP
     WLP* mp_WLP;
 
-    //!Pointer to the TypeLookupManager
-    fastdds::dds::builtin::TypeLookupManager* tlm_;
+    //! TypeLookupManager
+    fastdds::dds::builtin::TypeLookupManager typelookup_manager_;
```
Comment (richiware) [inline test/mock/rtps/BuiltinProtocols/fastdds/rtps/builtin/BuiltinProtocols.h:81]: Why this change is here? It's already on `feature/xtypes1.3` branch?

**Review by richiware — CHANGES_REQUESTED** [review]

File: test/mock/rtps/BuiltinProtocols/fastdds/rtps/builtin/BuiltinProtocols.h (hunk 8 above)
Comment (JLBuenoLopez-eProsima) [reply test/mock/rtps/BuiltinProtocols/fastdds/rtps/builtin/BuiltinProtocols.h:81]: Yes, this is something already merged in Fast DDS XTypes development branch but it is going to be reverted. But until then, in order to build the tests using this mock, the change is required.

File: test/blackbox/types/HelloWorldTypeObject.cxx (hunk 5 above)
Comment (JLBuenoLopez-eProsima) [reply test/blackbox/types/HelloWorldTypeObject.cxx:116]: This file is going to be removed when the TypeObject is generated in my integration branch. I would not be worried.

File: src/cpp/statistics/types/monitorservice_typesCdrAux.hpp (hunk 4 above)
Comment (JLBuenoLopez-eProsima) [reply src/cpp/statistics/types/monitorservice_typesCdrAux.hpp:45]: Looking at the IDL file, the MonitorServiceStatusData is a structure which has a union member. This union has a member which is a sequence of a structure (Connection) that contains two more sequences. Consequently, the size reduction would be caused by the change in behavior in Fast DDS-Gen of allocating only 50 elements in unbounded collections, instead of 100.

File: src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp:157]: That structure is annotated in the IDL file as final. Nevertheless, that specific structure is used for Strongly Connected Components which are not yet supported.

File: include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobjectCdrAux.hpp (hunk 1 above)
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/dds/xtypes/type_representation/detail/dds_xtypes_typeobjectCdrAux.hpp:27]: This can also be caused for allocating less samples in unbounded collections

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Review by richiware — DISMISSED** [review]

**Review by adriancampo — CHANGES_REQUESTED** [review]
Do not ask me how i know but...
Gives errors with -DFASTDDS_STATISTICS=OFF

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

**Review by richiware — APPROVED** [review]

**Comment by richiware** [comment]
> Do not ask me how i know but... Gives errors with -DFASTDDS_STATISTICS=OFF

Good catch

**Review by adriancampo — APPROVED** [review]
